### PR TITLE
Update packaging config to allow installation via brooklyn.libraries

### PR DESCRIPTION
**Pull request by tbouron**
To install a bundle via a bom file and brooklyn.libraries, you need to have:

- a valid OSGi bundle (was already there)
- a `catalog.bom` at the root of the bundle, which this PR adds

It also reshuffle a bit the bom files to make internal boms use` brooklyn.libraries` when necessary

### CODE REVIEW

File: common/catalog/catalog.bom [hunk 1]
```diff
@@ -22,4 +23,14 @@ brooklyn.catalog:
       iconUrl: classpath://io.brooklyn.clocker.common:icons/haproxy.png
       item:
         services:
-          - type: haproxy-load-balancer
\ No newline at end of file
+          - type: haproxy-load-balancer
+
+    - id: docker-engine-template
```
Comment (m4rkmckenna) [inline common/catalog/catalog.bom]: Can you remove this template ... 
IMO its just wrapping the docker-engine entity adding no extra value

File: kubernetes/catalog/kubernetes/kubernetes.bom [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+  brooklyn.libraries:
+  - mvn:io.brooklyn.clocker/clocker-common
+  - mvn:io.brooklyn.etcd/brooklyn-etcd
+
   items:
-  - classpath://io.brooklyn.etcd.brooklyn-etcd:etcd/etcd.bom
-  - classpath://io.brooklyn.clocker.common:docker/docker.bom
-  - classpath://io.brooklyn.clocker.common:common/haproxy.bom
-  - classpath://io.brooklyn.clocker.kubernetes:kubernetes/brooklyn-dns-etc-hosts-generator.bom
+  - classpath://kubernetes/brooklyn-dns-etc-hosts-generator.bom
```
Comment (m4rkmckenna) [inline kubernetes/catalog/kubernetes/kubernetes.bom]: The way this bundle makes it way into clocker seems hacky now https://github.com/brooklyncentral/clocker/blob/master/kubernetes/pom.xml#L39-L56

Can you update it as part of this PR please

File: common/catalog/catalog.bom (hunk 1 above)
Comment (tbouron) [reply common/catalog/catalog.bom]: Sure, I'll remove it

**Comment by tbouron** [comment]
@m4rkmckenna Comments addressed. Although, this now requires https://github.com/brooklyncentral/brooklyn-dns/pull/2 and we probably need to build and push it to maven central before this PR

**Comment by Graeme-Miller** [comment]
Merged. Marks comments have been addressed

**Pull request merged**